auth/src/main/java/org/aerogear/android/ags/auth/utils/AuthStateManager.java

```diff
@@ -1,60 +1,68 @@
-package org.aerogear.auth.utils;
+package org.aerogear.android.ags.auth.utils;
 
 import android.content.Context;
 import android.content.SharedPreferences;
-import org.aerogear.auth.credentials.OIDCCredentials;
-import org.json.JSONException;
+
+import org.aerogear.android.ags.auth.credentials.OIDCCredentials;
 
 /**
  * Saves, retrieves and delete a token.
  */
 public class AuthStateManager {
 
+    private static AuthStateManager instance = null;
     private static final String STORE_NAME = "org.aerogear.android.auth.AuthState";
     private static final String KEY_STATE = "state";
 
     private final SharedPreferences prefs;
 
-    public AuthStateManager(final Context context) {
+    private AuthStateManager(final Context context) {
```

Comment:
I'm not sure if Singleton's are suppose to have parameters? This way couldn't the singleton be instantiated multiple times which defeats the purpose?

Reply:
Private constructor and prefs is a final variable.  IN this case it isn't bad because the references set up aren't mutable.  However I am generally -1 to static singletons because they have annoying behaviors when you are testing that you have to code around.